lib/miq_pglogical.rb

```diff
@@ -1,122 +1,48 @@
 require 'pg'
-require 'pg/pglogical'
-require 'pg/pglogical/active_record_extension'
+require 'pg/logical_replication'
 
 class MiqPglogical
   include Vmdb::Logging
 
-  REPLICATION_SET_NAME = 'miq'.freeze
-  NODE_PREFIX = "region_".freeze
+  PUBLICATION_NAME = 'miq'.freeze
   ALWAYS_EXCLUDED_TABLES = %w(ar_internal_metadata schema_migrations repl_events repl_monitor repl_nodes).freeze
 
-  attr_reader :configured_excludes
-
   def initialize
-    @connection = ApplicationRecord.connection
-    self.configured_excludes = provider? ? active_excludes : self.class.default_excludes
-  end
-
-  # Sets the tables that should be used to create the replication set using refresh_excludes
-  def configured_excludes=(new_excludes)
-    @configured_excludes = new_excludes | ALWAYS_EXCLUDED_TABLES
+    @pg_connection = ApplicationRecord.connection.raw_connection
```

Comment:
I was thinking about this, and could this create issues with ActiveRecord connection pooling?

Reply:
If this were to checkout a new connection I guess it could, but otherwise it's using the same connection that the worker process would be using anyway, right? I guess the exception would be if we're using any features provided specifically by AR like locking or something, but if you think it could be an issue I can grab the conn info out of this object and open up a new connection. What do you think?

Reply:
I think I'm just concerned that we are holding a handle to the raw connection, which may also be given out again when someone else calls ApplicationRecord.connection.  I'm not sure if 2 threads using the same raw_connection is a problem or not.  When we were using the ActiveRecord connection instead of the raw connection I was less concerned because that was the object being pooled. (TBH, I don't understand how the pooling works when you call ApplicationRecord.connection, since I don't know what puts it back into the pool)

Reply:
Yeah, definitely a valid concern. @jrafanie any thoughts here?

Reply:
The pooling is thread based.  My concern would be that we consume the connection for the life of this client and could cause us to exhaust the pool if these clients could ever run at the same time.  Additionally, I don't think `connection` will explicitly checkin the connection so I'm not sure when this would be checked in, otherwise when the pool is full and the reaper starts checking for dead threads holding connections.

If you use the connection configuration to establish the connection with the PG gem directly, it feels like you'll get the parts of of activerecord you want without the risk of peeing in the pool.

Reply:
If it's thread based, and the lifetime of this MiqPglogical object is on-demand, then we are probably ok and can merge.  On the other hand, if we keep a cached instance of the MiqPglogical instance, then we would leak the underlying connection, and that would be an issue.

Reply:
Yeah, that's what I was saying before about checking out a new connection. But I think it's just using the one that's already assigned to the thread.

And as far as I can tell the only place that's even saving an instance past one invocation is a rake task:
```
app/models/miq_region.rb:146:    MiqPglogical.new.provider?
app/models/miq_region.rb:150:    MiqPglogical.new.subscriber?
app/models/miq_region.rb:167:    MiqPglogical.new.destroy_provider   if current_type == :remote
app/models/miq_region.rb:169:    MiqPglogical.new.configure_provider if desired_type == :remote
lib/tasks/evm_dbsync.rake:24:      pgl = MiqPglogical.new
```